src/pages/home/report/ReportActionItem.tsx

```diff
@@ -760,75 +785,54 @@ function ReportActionItem(props) {
                 )}
             </Hoverable>
             <View style={styles.reportActionSystemMessageContainer}>
-                <InlineSystemMessage message={props.action.error} />
+                {/* @ts-expect-error TODO check if there is a field on the reportAction object */}
+                <InlineSystemMessage message={action.error} />
             </View>
         </PressableWithSecondaryInteraction>
     );
 }
 
-ReportActionItem.propTypes = propTypes;
-ReportActionItem.defaultProps = defaultProps;
-
-export default compose(
-    withWindowDimensions,
-    withLocalize,
-    withNetwork(),
-    withBlockedFromConcierge({propName: 'blockedFromConcierge'}),
-    withReportActionsDrafts({
-        propName: 'draftMessage',
-        transformValue: (drafts, props) => {
-            const originalReportID = ReportUtils.getOriginalReportID(props.report.reportID, props.action);
-            const draftKey = `${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}${originalReportID}`;
-            return lodashGet(drafts, [draftKey, props.action.reportActionID, 'message']);
-        },
-    }),
-    withOnyx({
-        preferredSkinTone: {
-            key: ONYXKEYS.PREFERRED_EMOJI_SKIN_TONE,
-            initialValue: CONST.EMOJI_DEFAULT_SKIN_TONE,
-        },
-        iouReport: {
-            key: ({action}) => {
-                const iouReportID = ReportActionsUtils.getIOUReportIDFromReportActionPreview(action);
-                return iouReportID ? `${ONYXKEYS.COLLECTION.REPORT}${iouReportID}` : undefined;
-            },
-            initialValue: {},
-        },
-        policyReportFields: {
-            key: ({report}) => (report && 'policyID' in report ? `${ONYXKEYS.COLLECTION.POLICY_REPORT_FIELDS}${report.policyID}` : undefined),
-            initialValue: [],
-        },
-        emojiReactions: {
-            key: ({action}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS_REACTIONS}${action.reportActionID}`,
-            initialValue: {},
-        },
-        userWallet: {
-            key: ONYXKEYS.USER_WALLET,
-        },
-        parentReportActions: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.parentReportID || 0}`,
-            canEvict: false,
+export default withOnyx<ReportActionItemProps, ReportActionItemOnyxProps>({
+    preferredSkinTone: {
+        key: ONYXKEYS.PREFERRED_EMOJI_SKIN_TONE,
+        initialValue: CONST.EMOJI_DEFAULT_SKIN_TONE,
+    },
+    iouReport: {
+        key: ({action}) => {
+            const iouReportID = ReportActionsUtils.getIOUReportIDFromReportActionPreview(action);
+            return `${ONYXKEYS.COLLECTION.REPORT}${iouReportID}`;
         },
-    }),
-)(
+    },
+    policyReportFields: {
+        key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY_REPORT_FIELDS}${report.policyID}`,
```

Comment:
I think its working the same way because now even if there is no `policyID` it will return undefined

Reply:
I'm not sure actually, because in the old logic the entire `key` will be `undefined` if `report` or `report.policyID` don't exist, while in the new logic `key` will be something like `policyReportFields_undefined`. Please test this case if it makes difference during runtime

Reply:
Yes I tested that and I just passed `policyReportFields_undefined` hardcoded into return of that key and in the props we had `policyReportFields: undefined` , so basically the same thing which was earlier 😄